eureka-client/src/main/java/com/netflix/eureka2/client/channel/SnapshotInterestChannelImpl.java

```diff
@@ -0,0 +1,101 @@
+package com.netflix.eureka2.client.channel;
+
+import com.netflix.eureka2.client.transport.tcp.TcpDiscoveryClient;
+import com.netflix.eureka2.interests.Interest;
+import com.netflix.eureka2.protocol.discovery.AddInstance;
+import com.netflix.eureka2.protocol.discovery.SnapshotComplete;
+import com.netflix.eureka2.protocol.discovery.SnapshotRegistration;
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+import com.netflix.eureka2.transport.MessageConnection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+import rx.functions.Action0;
+import rx.functions.Action1;
+import rx.functions.Func1;
+
+/**
+ * @author Tomasz Bak
+ */
+public class SnapshotInterestChannelImpl implements SnapshotInterestChannel {
+
+    private static final Logger logger = LoggerFactory.getLogger(SnapshotInterestChannel.class);
+
+    private final TcpDiscoveryClient transportClient;
+
+    public SnapshotInterestChannelImpl(TcpDiscoveryClient transportClient) {
+        this.transportClient = transportClient;
+    }
+
+    @Override
+    public Observable<InstanceInfo> forSnapshot(final Interest<InstanceInfo> interest) {
+        return transportClient.connect()
+                .flatMap(new Func1<MessageConnection, Observable<InstanceInfo>>() {
+                    @Override
+                    public Observable<InstanceInfo> call(final MessageConnection connection) {
+                        return connection
+                                // Subscribe to interest set
+                                .submitWithAck(new SnapshotRegistration(interest))
+                                .cast(Object.class)
+                                .concatWith(connection.incoming())
+                                        // Handle change notification stream
+                                .takeWhile(END_OF_STREAM_FILTER)
+                                .filter(INVALID_MESSAGE_FILTER)
+                                .doOnNext(new Action1<Object>() {
+                                    @Override
+                                    public void call(Object message) {
+                                        connection.acknowledge().subscribe();
```

Comment:
Now we only log error messages if acks fail. I will add logging here.
In the future we will use acks for back pressure.